### PR TITLE
Use `copied` instead of manual `map`

### PR DESCRIPTION
**Pull request by DaniPopes**

### CODE REVIEW

File: compiler/rustc_ast_lowering/src/expr.rs [hunk 1]
```diff
@@ -41,8 +41,7 @@ impl<'hir> LoweringContext<'_, 'hir> {
                     }
                     // Merge attributes into the inner expression.
                     if !e.attrs.is_empty() {
-                        let old_attrs =
-                            self.attrs.get(&ex.hir_id.local_id).map(|la| *la).unwrap_or(&[]);
+                        let old_attrs = self.attrs.get(&ex.hir_id.local_id).copied().unwrap_or(&[]);
```
Comment (petrochenkov) [inline compiler/rustc_ast_lowering/src/expr.rs:44]:
```suggestion
                        let old_attrs = self.attrs.get(&ex.hir_id.local_id).copied().unwrap_or_default();
```

Comment (eggyal) [reply compiler/rustc_ast_lowering/src/expr.rs:44]: I don't feel strongly about it, but very slightly prefer `unwrap_or`: it's both fewer keystrokes and slightly less cognitive load for the reader (no need to consider the type or what its `Default` implementation is).

Comment (DaniPopes) [reply compiler/rustc_ast_lowering/src/expr.rs:44]: Both are present in the codebase, I don't think this should be changed, but if anything I think it should be in a follow-up PR.

**Comment by nnethercote** [comment]
How did you find these? Is it a clippy lint?

**Comment by DaniPopes** [comment]
> How did you find these? Is it a clippy lint?

Apparently it is one: https://rust-lang.github.io/rust-clippy/master/#/map_clone
But I just used a regex: `\.map\(\|[^,]+\| \*\w+\)`

**Pull request merged**